cdk/test/functions/service-control-policies-root.test.ts

```diff
@@ -0,0 +1,252 @@
+          Id: logicalResourceId,
+          scpName: scpName,
+        },
+      } as unknown as CloudFormationCustomResourceCreateEvent,
+      {} as Context,
+    );
+
+    expect(response).toBeTruthy();
+  });
+
+  it('no root account found in organization', async () => {
+    organizationClientMock.on(ListRootsCommand).resolves({
+      Roots: [],
+    });
+
+    try {
+      await handler(
+        {
+          RequestType: 'Create',
+          ResourceProperties: {
+            Type: PolicyType.SERVICE_CONTROL_POLICY,
+            Description: logicalResourceId,
+            scpName: scpName,
+            Content: updatedSCPPolicy,
+          },
+        } as unknown as CloudFormationCustomResourceCreateEvent,
+        {} as Context,
+      );
+    } catch (e) {
+      expect(e).toMatchObject(new Error('No root account found in the organization'));
+    }
+  });
+
+  it('no matching policy found', async () => {
+    //Should throw an error when there are no policies matching the name
+    organizationClientMock.on(ListPoliciesCommand).resolves({
+      Policies: [
+        {
+          Name: scpName,
+          Id: logicalResourceId,
+          Description: description,
+        },
+      ],
+    });
+
+    organizationClientMock.on(UpdatePolicyCommand).resolves({
+      Policy: {
+        Content: '{"Version": "2012-10-17", "Statement": [/* UPDATED policy document */]}',
+        PolicySummary: {
+          Id: policyId,
+          Name: scpName,
+        },
+      },
+    });
+
+    try {
+      await handler(
+        {
+          RequestType: 'Update',
+          ResourceProperties: {
+            Type: PolicyType.SERVICE_CONTROL_POLICY,
+            Description: logicalResourceId,
+            scpName: 'mock-scp-name', //should throw an error since there will be no matching policy for this name
+            Content: updatedSCPPolicy,
+          },
+        } as unknown as CloudFormationCustomResourceCreateEvent,
+        {} as Context,
+      );
+    } catch (e) {
+      expect(e).toMatchObject(new Error('No SCP Policy found for the name: mock-scp-name'));
+    }
+  });
+
+  it('delete failed for scp when DetachPolicyCommand fails', async () => {
+    organizationClientMock.on(ListRootsCommand).resolves({});
+
+    organizationClientMock.on(ListPoliciesCommand).resolves({
+      Policies: [
+        {
+          Name: scpName,
+          Id: logicalResourceId,
+          Description: description,
+        },
+      ],
+    });
+
+    //organizationClientMock.on(DetachPolicyCommand).resolves({});
+    organizationClientMock.on(DeletePolicyCommand).resolves({});
+
+    try {
+      await handler(
+        {
+          RequestType: 'Delete',
+          ResourceProperties: {
+            Id: logicalResourceId,
+          },
+        } as unknown as CloudFormationCustomResourceCreateEvent,
+        {} as Context,
+      );
+    } catch (e) {
+      expect(e).toMatchObject(new Error('No root account found in the organization'));
```

Comment:
I belive that's not the correct error that is supposed to appear on a failed delete